design.md

```diff
@@ -0,0 +1,70 @@
+# Design
+
+react-conf-webrtc is designed to be flexible enough to support your signaling system. In order to accomplish this
+react-conf-webrtc uses an event system with well-defined messages that you can send and expect to receive.
+
+Out of the box, react-conf-webrtc ships with an adapter for the Spreed RTC signaling events. This adapter acts
+as a middleman between the Spreed WebSocket server events and the conference room and translates them to be
+compatible with the react-conf-webrtc API and vice versa.
+
+The SpreedAdapter internally will queue the messages if there are no messages available. This is necessary because
```

Comment:
SpreedAdapter is an adapter between Spreed API events and react-conf-webrtc API. For incoming signaling events, it will translate them into the react-conf-webrtc API. For outgoing message, it will do the opposite. It will translate react-conf-webrtc messages into Spreed signaling events.